File: python/difference-of-squares/difference_of_squares.py
def natural_range(number: int) -> list[int]:
    return range(1, number + 1)


def square_of_sum(number: int) -> int:
    return sum(natural_range(number)) ** 2


def sum_of_squares(number: int) -> int:
    return sum([n ** 2 for n in natural_range(number)])


def difference_of_squares(number: int) -> int:
    return square_of_sum(number) - sum_of_squares(number)

File: scala/difference-of-squares/src/main/scala/DifferenceOfSquares.scala
object DifferenceOfSquares {
  def sumOfSquares(n: Int): Int = (1 to n).map(square).sum

  def squareOfSum(n: Int): Int = (square _).apply((1 to n).sum)

  def differenceOfSquares(n: Int): Int = squareOfSum(n) - sumOfSquares(n)

  private def square(n: Int) = n * n
}
